protocols/gossipsub/src/handler.rs

```diff
@@ -475,14 +438,14 @@ impl ConnectionHandler for Handler {
                                         Some(OutboundSubstreamState::WaitingOutput(substream));
                                 }
                                 Err(e) => {
-                                    error!("Error sending message: {}", e);
-                                    return Poll::Ready(ConnectionHandlerEvent::Close(e));
+                                    error!("Error sending message: {e}");
```

Comment:
I think we also need to set `outbound_substream` to `None` here otherwise we remain in "Poisoned".

Reply:
Previously, we returned `ConnectionHandlerEvent::Close` which meant we would never call `poll` again on this handler. This is changing now so we need to update the code accordingly so those state machines are never in poisoned.